packages/process/src/node/raw-process.ts

```diff
@@ -116,27 +116,50 @@ export class RawProcess extends Process {
                 this.emitOnError(error as ProcessErrorEvent);
             });
             this.process.on('exit', (exitCode: number, signal: string) => {
+
+            });
+
+            // When no stdio option is passed, it is null by default.
+            this.outputStream = this.process.stdout || new DevNullStream({ autoDestroy: true });
+            this.inputStream = this.process.stdin || new DevNullStream({ autoDestroy: true });
+            this.errorStream = this.process.stderr || new DevNullStream({ autoDestroy: true });
+
+            const onExit = new Promise<{ code?: number, signal?: string }>(resolve => {
+                this.process!.on('exit', (code, signal) => resolve({ code, signal }));
+            });
+            const onClose = new Promise(resolve => {
+                this.process!.on('close', () => resolve());
+            });
+
+            onExit.then(exitStatus => {
+                const { code: exitCode, signal } = exitStatus;
                 // node's child_process exit sets the unused parameter to null,
                 // but we want it to be undefined instead.
                 this.emitOnExit(
-                    exitCode !== null ? exitCode : undefined,
-                    signal !== null ? signal : undefined,
+                    typeof exitCode === 'number' ? exitCode : undefined,
+                    typeof signal === 'string' ? signal : undefined,
                 );
             });
 
-            this.outputStream = this.process.stdout || new DevNullStream();
-            this.inputStream = this.process.stdin || new DevNullStream();
-            this.errorStream = this.process.stderr || new DevNullStream();
+            Promise.all([onExit, onClose]).then(resolved => {
```

Comment:
I still don't get it both `close` and `exit` events already provide code and signal. Should not we simply translate them? Why do we need to check semantic defined by Node.js?

i.e.
```ts
this.process.on('exit', (code, signal) => this.emitOnExit({ code, signal }));
this.process.on('close', (code, signal) => this.emitOnClose({ code, signal }));
```

Reply:
These are left overs from when I was listening to the streams.